[FE-MAP-01, FE-MAP-05] 지도, FAB 버튼

Daram0o0 opened the pull request:
## 📌 관련 이슈 (Task ID)

- ID: FE-MAP-01, FE-MAP-05

## 📝 작업 내용 (Details)

1. 메인 지도 화면 구현 (FE-MAP-01)
- [x] react-native-webview를 활용한 카카오맵 전체 화면 구현
- [x] 하단 탭바(Bottom Tab) 레이아웃 배치 및 지도 탭 연동
- [ ] 지도 로딩 상태 처리 (ActivityIndicator) 및 WebView 설정 적용

2. FAB(Floating Action Button) 인터랙션 구현 (FE-MAP-05)
- [x] 우측 하단 메인 FAB 버튼 컴포넌트 개발 (commons/components/fab-button)
- [x] 버튼 클릭 시 확장 애니메이션 구현
   - 배경 딤(Dim) 처리 (Overlay)
   - 메인 아이콘 변경 (+ ↔ X)
   - 하위 메뉴 ('타임캡슐', '이스터에그') 노출 애니메이션 적용
- [ ] 각 메뉴 클릭 시 해당 라우팅 경로 연결

## 📸 스크린샷 (Screenshots)

<img width="421" height="1017" alt="스크린샷 2025-12-16 21 47 24" src="https://github.com/user-attachments/assets/ad2af373-e200-42ff-af5f-4326e2cb1581" />
<img width="421" height="1017" alt="스크린샷 2025-12-16 21 47 24" src="https://github.com/user-attachments/assets/ad2af373-e200-42ff-af5f-4326e2cb1581" />

## 🧐 리뷰 포인트 (Review Point)

- 지도 구현 방식
- FAB 애니메이션
- 컴포넌트 분리

## ✅ 체크리스트 (Checklist)

- [x] 컨벤션(커밋, 브랜치 네이밍)을 준수했나요?
- [ ] 불필요한 콘솔 로그(console.log)나 주석을 제거했나요?
- [x] 실행 시 에러가 발생하지 않나요?

Daram0o0 committed:
chore: app.json을 app.config.js로 마이그레이션

환경 변수(.env) 사용을 위해 정적 app.json을 동적 app.config.js로 변경

Daram0o0 committed:
chore: @react-navigation/drawer 의존성 제거

Daram0o0 committed:
docs: 패키지 문서 업데이트 및 react-native-webview 추가

- react-native-webview v13.16.0 추가
- @react-navigation/drawer 문서에서 제거 (미설치 패키지)
- 패키지 번호 재정렬 및 카테고리별 분류 업데이트

Daram0o0 committed:
feat: 카카오 지도 WebView 렌더링 구현

- WebView를 사용한 카카오 지도 렌더링 기능 추가
- expo-constants를 통한 카카오 API 키 주입
- 양방향 메시지 통신 구현 (INIT, READY, MAP_CLICK, MARKER_CLICK)
- 마커 추가/제거 및 카메라 이동 기능 지원
- 초기 위치 서울시청으로 설정

Daram0o0 committed:
feat: 홈 화면에 FAB 버튼 컴포넌트 추가

- FAB 버튼 컴포넌트 구현 (commons/components/fab-button)
- 메인 버튼 클릭 시 이스터에그/타임캡슐 서브 메뉴 확장
- React Native Reanimated를 활용한 애니메이션 효과
- 우측 하단 고정 위치 배치
- 접근성 속성 완비 (accessibilityRole, accessibilityLabel)
- SVG 아이콘 에셋 추가 (plus, close, egg, capsule)
- HomePage에 FAB 버튼 통합 및 이벤트 핸들러 구현

Daram0o0 committed:
refactor: 지도 기능을 Feature Slice 구조로 재구성

- map feature를 완전한 Feature Slice 구조로 재조정
- FAB 버튼을 commons에서 map feature 내부로 이동
- WebView 관련 코드를 map-view 컴포넌트로 분리
- hooks, components, types 계층 구조 명확화
- utils, commons/layout/provider 폴더 추가
- 폴더 구조 문서화 (folder-structure.md)

kimchiro commented on commons/layout/provider/README.md:
함수 UI 분리 해주세요.

kimchiro commented on commons/layout/Tabs/HomePage/index.tsx:
이동할 페이지 상수로 관리한후 페이지 이동 함수 공통 hooks로 생성후 ,  import 해서 사용해주세요.

11줄, handleEasterEggPress
16줄 handleTimeCapsulePress 

기능이 사실 같은 기능일텐데 합치는게 좋아보여요, 다른 사람도 사용할 가능성 높은 기능 같네요

kimchiro reviewed (changes requested):
리뷰 기반 수정후 다시 PR 올려주세요

Daram0o0 committed:
refactor: FAB 버튼 애니메이션 로직을 커스텀 훅으로 분리

- useFabAnimation 훅 생성으로 상태 관리 및 애니메이션 로직 추출
- FAB 컴포넌트 가독성 및 재사용성 향상
- UI 렌더링과 비즈니스 로직 분리로 유지보수성 개선

Daram0o0 committed:
Merge branch 'dev' into feat/map
